libmamba/include/mamba/core/channel.hpp

```diff
@@ -46,9 +45,11 @@ namespace mamba
         const std::string& name() const;
         const std::string& canonical_name() const;
         const std::vector<std::string>& platforms() const;
-        const std::optional<std::string>& auth() const;
-        const std::optional<std::string>& token() const;
-        const std::optional<std::string>& package_filename() const;
+        const std::optional<std::string> auth() const;
+        const std::optional<std::string> user() const;
+        const std::optional<std::string> password() const;
+        const std::optional<std::string> token() const;
+        const std::optional<std::string> package_filename() const;
```

Comment:
No need for `const` for objects returned by value.

Reply:
Good catch.